### PR TITLE
Created new plugin for LSO

### PR DESCRIPTION
**Pull request by gnehapk**
This new plugin is required to create UI for -

- automated discovery of local disks via the `LocalVolumeDiscovery` CR and exposing the result via `LocalVolumeNodeDiscoveryResult` CR.
- automated provisioning`VolumeDiscovery` CRD and sorting them into StorageClasses based on their characteristics via the `LocalVolumeGroup` CRD. 

These CRDs will be implemented in the LocalStorageOperator(LSO)

This is required to automate OpenShift Container Storage deployment on bare metal from UI.
More details can be found [here](https://github.com/openshift/enhancements/pull/237)

### CODE REVIEW

File: frontend/packages/local-storage-operator-plugin/src/models.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  labelPlural: 'Volumes',
+  apiVersion: 'v1alpha1',
+  apiGroup: 'local.storage.openshift.io',
+  plural: 'volumes',
+  abbr: 'ADV',
+  namespaced: true,
+  kind: 'AutoDetectVolume',
+  id: 'autodetectvolume',
+  crd: true,
+};
```
Comment (bipuladh) [inline frontend/packages/local-storage-operator-plugin/src/models.ts]: new line

File: frontend/packages/local-storage-operator-plugin/package.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "@console/local-storage-operator-plugin",
+  "version": "0.0.0-fixed",
+  "description": "LSO - Operator for local storage",
+  "private": true,
+  "dependencies": {
+    "@console/plugin-sdk": "0.0.0-fixed",
+    "@console/shared": "0.0.0-fixed"
```
Comment (bipuladh) [inline frontend/packages/local-storage-operator-plugin/package.json]: Add `console/internal`

File: frontend/packages/local-storage-operator-plugin/OWNERS [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  - gnehapk
+  - rawagner
+approvers:
+  - afreen23
+  - bipuladh
+  - cloudbehl
+  - gnehapk
+  - rawagner
+labels:
+  - component/lso
```
Comment (bipuladh) [inline frontend/packages/local-storage-operator-plugin/OWNERS]: You need to send a separate PR for adding this label. Add the link to the description to this PR.

File: frontend/packages/local-storage-operator-plugin/OWNERS [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  - bipuladh
+  - cloudbehl
+  - gnehapk
+  - rawagner
+approvers:
+  - afreen23
+  - bipuladh
+  - cloudbehl
+  - gnehapk
+  - rawagner
```
Comment (bipuladh) [inline frontend/packages/local-storage-operator-plugin/OWNERS]: No need to add him. He's the OWNER of the whole console.

File: frontend/packages/local-storage-operator-plugin/OWNERS (hunk 3 above)
Comment (gnehapk) [reply frontend/packages/local-storage-operator-plugin/OWNERS]: Why is it so?

File: frontend/packages/local-storage-operator-plugin/OWNERS (hunk 4 above)
Comment (gnehapk) [reply frontend/packages/local-storage-operator-plugin/OWNERS]: Since, he was added in ceph and nooba plugin, I followed the same.

Comment (bipuladh) [reply frontend/packages/local-storage-operator-plugin/OWNERS]: Back then he was not.

File: frontend/packages/local-storage-operator-plugin/OWNERS (hunk 3 above)
Comment (bipuladh) [reply frontend/packages/local-storage-operator-plugin/OWNERS]: Putting this here is fine. But I am assuming you want the label support in GitHub PRs too right? For that, you need to raise a separate PR.

File: frontend/packages/local-storage-operator-plugin/src/plugin.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+} from '@console/plugin-sdk';
+import * as models from './models';
+
+type ConsumedExtensions =
+  | ModelFeatureFlag
+  | ModelDefinition
+  | ResourceNSNavItem
+  | ResourceListPage
+  | YAMLTemplate
+  | RoutePage;
```
Comment (afreen23) [inline frontend/packages/local-storage-operator-plugin/src/plugin.ts]: These arent used anywhere.

File: frontend/packages/local-storage-operator-plugin/src/models.ts (hunk 1 above)
Comment (afreen23) [reply frontend/packages/local-storage-operator-plugin/src/models.ts]: Whats the difference in `LocalVolumeDiscovery` CR and this one ?
Arent we supposed to add that ?

Comment (gnehapk) [reply frontend/packages/local-storage-operator-plugin/src/models.ts]: `LocalVolumeDiscovery` is required to auto detect the local devices on nodes, which is not part of this release 4.5.

File: frontend/packages/local-storage-operator-plugin/OWNERS (hunk 3 above)
Comment (gnehapk) [reply frontend/packages/local-storage-operator-plugin/OWNERS]: Got it. Thanks!

File: frontend/packages/local-storage-operator-plugin/src/models.ts (hunk 1 above)
Comment (cloudbehl) [inline frontend/packages/local-storage-operator-plugin/src/models.ts]: Use `LocalVolume` CR to enable a plugin , not the AutoDetectVolume

**Comment by gnehapk** [comment]
@cloudbehl @bipuladh Please review.

File: frontend/packages/local-storage-operator-plugin/src/models.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  labelPlural: 'Local Volumes',
+  apiVersion: 'v1',
+  apiGroup: 'local.storage.openshift.io',
+  plural: 'localvolumes',
+  abbr: 'LV',
+  namespaced: true,
+  kind: 'LocalVolume',
+  id: 'localvolume',
+  crd: true,
+};
```
Comment (gnehapk) [inline frontend/packages/local-storage-operator-plugin/src/models.ts]: @ashishranjan738 Can you please review the models as well for `AutoDetectVolumeModel` and `LocalVolumeModel`

Comment (ashishranjan738) [reply frontend/packages/local-storage-operator-plugin/src/models.ts]: @gnehapk There is nothing called `AutoDetectVolume` CR any more it has been renamed to `LocalVolumeGroup`. You can find the finalized designs for it here https://github.com/openshift/enhancements/blob/master/enhancements/local-storage/automatic-detection.md.

Comment (ashishranjan738) [reply frontend/packages/local-storage-operator-plugin/src/models.ts]: The `LocalVolume` Model seems to okay

**Comment by cloudbehl** [comment]
/lgtm cancel

This needs to be addressed. https://github.com/openshift/console/pull/4759#pullrequestreview-377655020

**Comment by gnehapk** [comment]
@cloudbehl @ashishranjan738 Updated the PR. Please review
